compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -1451,6 +1451,26 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
                         .collect();
 
                     if non_visible_spans.len() > 0 {
+                        if let Some(visibility_spans) = self.r.field_visibility_spans.get(&def_id) {
+                            err.multipart_suggestion_verbose(
+                                &format!(
+                                    "consider making the field{} publicly accessible",
+                                    pluralize!(visibility_spans.len())
+                                ),
+                                visibility_spans
+                                    .iter()
+                                    .map(|span| {
+                                        (
+                                            *span,
+                                            if span.is_empty() { "pub " } else { "pub" }
```

Comment:
This could lead to an invalid suggestion if the user wrote something like this
```rust
mod a {
    pub struct A(pub(self)String);
}

fn main() {
    a::A("a".into());
}
```
I'm not sure how likely this is to come up (since it's non-standard formatting) so I'll leave it up to you whether you want to fix it.

Reply:
Hm, my example actually doesn't trigger the error at all..
```rust
error[E0603]: tuple struct constructor `A` is private
 --> aa.rs:6:8
  |
2 |     pub struct A(pub(self)String);
  |                  --------------- a constructor is private if any of the fields is private
...
6 |     a::A("a".into());
  |        ^ private tuple struct constructor
  |
note: the tuple struct constructor `A` is defined here
 --> aa.rs:2:5
  |
2 |     pub struct A(pub(self)String);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

But this version does produce an invalid suggestion:
```rust
mod a {
    pub struct A(pub(self)String);
}

mod b {
    use crate::a::A;
    fn x() {
        A("".into());
    }
}

fn main() {}
```

```rust
help: consider making the field publicly accessible
  |
2 |     pub struct A(pubString);
  |                  ~~~
```

Reply:
Fixed